docs/requirements.in

```
@@ -1,6 +1,7 @@
 myst-parser[linkify] >= 0.15.1
 setuptools-scm >= 6.0.1
-Sphinx >= 4.1.2
+# https://github.com/sphinx-doc/sphinx/issues/10112
+Sphinx >= 4.1.2, < 4.4.0
```

It's best to not create detached comments where possible:
```suggestion
Sphinx >= 4.1.2, < 4.4.0  # https://github.com/sphinx-doc/sphinx/issues/10112
```